### PR TITLE
SLI-1710 sonarlint-intellij golden based image

### PR DESCRIPTION
**Pull request by julien-carsique-sonarsource**
Make use of https://github.com/SonarSource/sonarlint-ci-images/pull/24

- **SLI-1710 Use custom AMI Windows Image**
- **SLI-1710 use pr-24-sonarlint-intellij image**

Part of 

* Please review our [contribution guidelines](https://github.com/SonarSource/sonarlint-intellij/blob/master/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [x] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [x] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [x] Provide a unit test for any code you changed
- [x] If there is a [JIRA](http://jira.sonarsource.com/browse/SLI) ticket available, please make your commits and pull request start with the ticket ID (SLI-XXXX)

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -147,18 +146,19 @@ validate_windows_task:
...
   ec2_instance:
     <<: *WINVM_DEFINITION
   <<: *SETUP_GRADLE_CACHE
   # skip unit tests from submodules to allow caching (else cache is too big for Cirrus)
   build_script: |
     source cirrus-env QA
     source .cirrus/use-gradle-wrapper.sh
-    ./gradlew :check
+    mkdir "/C/Program Files/Microsoft/jdk-17.0.12.7-hotspot/Packages"
+    ./gradlew :check -x :its:check
```
Comment (eray-felek-sonarsource) [inline .cirrus.yml:161]: its:check runs other things such as checking licenses so we should run it, it will skip qas anyway

**Comment by julien-carsique-sonarsource** [comment]
Remove ccba6298f7b029c1b9ccb11eae410ba032447fda before merge

File: README.md [hunk 2]
```diff
@@ -127,11 +139,18 @@ To run against a specific IDE, the `runIdeDirectory` property can be used as suc
 Keep in mind that the `clean` task will wipe out the content of `build/idea-sandbox`,
 so you will need to repeat some setup steps for that instance, such as configuring the JDK.
 
-Whenever you change the dependency version, the previous versions are not deleted from the sandbox, and the JVM might not load the version that you expect.
+Whenever you change the dependency version, the previous versions are not deleted from the sandbox, and the JVM might not load the version
```
Comment (eray-felek-sonarsource) [inline README.md:142]: I think we should revert these line changes

File: rider/build.gradle.kts [hunk 3]
```diff
@@ -7,9 +7,11 @@ plugins {
 
 intellij {
     if (!riderHome.isNullOrBlank()) {
+        println("Using local installation of IntelliJ IDEA: $riderHome")
```
Comment (nquinquenel) [inline rider/build.gradle.kts:10]: Technically, it's not IntelliJ IDEA but Rider

File: clion/build.gradle.kts [hunk 4]
```diff
@@ -4,9 +4,11 @@ val clionHome: String? = System.getenv("CLION_HOME")
 intellij {
     plugins.set(listOf("com.intellij.clion", "com.intellij.cidr.base", "com.intellij.cidr.lang"))
     if (!clionHome.isNullOrBlank()) {
+        println("Using local installation of IntelliJ IDEA: $clionHome")
```
Comment (nquinquenel) [inline clion/build.gradle.kts:7]: Should be CLion and not IDEA

File: clion-resharper/build.gradle.kts [hunk 5]
```diff
@@ -3,9 +3,11 @@ val resharperHome: String? = System.getenv("RESHARPER_HOME")
 
 intellij {
     if (!resharperHome.isNullOrBlank()) {
+        println("Using local installation of IntelliJ IDEA: $resharperHome")
```
Comment (nquinquenel) [inline clion-resharper/build.gradle.kts:6]: Should be Rider and not IDEA

File: README.md [hunk 6]
```diff
@@ -127,11 +139,18 @@ To run against a specific IDE, the `runIdeDirectory` property can be used as suc
...
+that you expect.
 As the `clean` task may be inconvenient, an easier workaround is to delete the jars in the sandbox, for example with:
 
     find build/idea-sandbox/ -name '*.jar' -delete
 
+Build Images
+------------
+
+The image used to build the plugin is defined in the
+`https://github.com/SonarSource/sonarlint-ci-images/blob/master/.cirrus/sonarlint_intellij_ami_tasks.star` pipeline.
```
Comment (nquinquenel) [inline README.md:152]: It's a private repository, I'm not sure it belongs in the README of a public repo

**Review by nquinquenel — APPROVED** [review]
Great work! Just a few comments

**Review by eray-felek-sonarsource — APPROVED** [review]

**Pull request merged**